endpoints/openrtb2/video_auction.go

```diff
@@ -659,27 +654,32 @@ func (deps *endpointDeps) validateVideoRequest(req *openrtb_ext.BidRequestVideo)
 		}
 	}
 
-	if len(req.Video.Mimes) == 0 {
-		err := errors.New("request missing required field: Video.Mimes")
-		errL = append(errL, err)
-	} else {
-		mimes := make([]string, 0, 0)
-		for _, mime := range req.Video.Mimes {
-			if mime != "" {
-				mimes = append(mimes, mime)
+	if req.Video != nil {
+		if len(req.Video.MIMEs) == 0 {
+			err := errors.New("request missing required field: Video.Mimes")
+			errL = append(errL, err)
+		} else {
+			mimes := make([]string, 0, 0)
+			for _, mime := range req.Video.MIMEs {
+				if mime != "" {
+					mimes = append(mimes, mime)
+				}
+			}
+			if len(mimes) == 0 {
+				err := errors.New("request missing required field: Video.Mimes, mime types contains empty strings only")
+				errL = append(errL, err)
+			}
+			if len(mimes) > 0 {
```

Comment:
This check doesn't seem necessary. I don't the harm of an assignment of `req.Video.MIMEs` to an empty slice.

Reply:
Removed check here. The error of having all empty strings will still be caught regardless of if the value is copied over empty.